java/client/src/main/java/glide/connectors/resources/ThreadPoolResource.java

```diff
@@ -0,0 +1,23 @@
+package glide.connectors.resources;
+
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.unix.DomainSocketChannel;
+import lombok.Getter;
+import lombok.NonNull;
+
+/**
+ * Abstract base class that provides the EventLoopGroup and DomainSocketChannel to be used by the
+ * Netty protocol.
+ */
+@Getter
+public abstract class ThreadPoolResource {
+    private EventLoopGroup eventLoopGroup;
+    private Class<? extends DomainSocketChannel> domainSocketChannelClass;
+
+    public ThreadPoolResource(
+            @NonNull EventLoopGroup eventLoopGroup,
+            @NonNull Class<? extends DomainSocketChannel> domainSocketChannelClass) {
```

Comment:
why does the user need to choose a socket implementation if they only want to set their own threadpool in the client?

Reply:
It's a Netty.io requirement.  And the socket channel class needs to match with the ELG provided as they work hand-in-hand.  The good news, is that Netty.io is responsible for checking that they can co-exist.

Reply:
so, this binds the user-facing interface of the client to Netty? How will you be able to enable Windows support without breaking backwards compatibility if the public APi is Netty-bound?

Reply:
We believe adding support for a nio will allow for Windows users.

Reply:
are the types used in nio compatible with the netty types?